### PR TITLE
Tuple implementation

### PR DESCRIPTION
**Pull request by kennethlove**
Adds `__pow__`, `__add__`, `__mod__`, and `__getitem__` for the `Tuple` class.

References #46

### CODE REVIEW

File: batavia/types/Tuple.js [hunk 1]
```diff
@@ -143,19 +143,70 @@ batavia.types.Tuple = function() {
...
 
     Tuple.prototype.__mod__ = function(other) {
-        throw new batavia.builtins.NotImplementedError("Tuple.__mod__ has not been implemented");
+        throw new batavia.builtins.TypeError("unsupported operand type(s) for %: 'tuple' and '" + batavia.type_name(other) + "'");
     };
 
     Tuple.prototype.__add__ = function(other) {
-        throw new batavia.builtins.NotImplementedError("Tuple.__add__ has not been implemented");
+		if (!batavia.isinstance(other, batavia.types.Tuple)) {
+			throw new batavia.builtins.TypeError('can only concatenate tuple (not "' + batavia.type_name(other) + '") to tuple')
```
Comment (freakboy3742) [inline batavia/types/Tuple.js]: Not sure how the tabs slipped in here...

File: tests/datatypes/test_tuple.py [hunk 2]
```diff
@@ -306,20 +266,20 @@ class BinaryTupleOperationTests(BinaryOperationTestCase, TranspileTestCase):
         'test_subscr_bool',
         'test_subscr_bytearray',
         'test_subscr_bytes',
-        'test_subscr_class',
+        # 'test_subscr_class',
```
Comment (freakboy3742) [inline tests/datatypes/test_tuple.py]: Delete these lines rather than commenting.

File: batavia/types/Tuple.js [hunk 3]
```diff
@@ -143,19 +143,70 @@ batavia.types.Tuple = function() {
...
+			for (i=0; i < this.length; i++){
+				result.push(this[i]);
+			}
+
+			for (i=0; i < other.length; i++){
+				result.push(other[i]);
+			}
+
+			return result;
+			// throw new batavia.builtins.NotImplementedError("Tuple.__add__ has not been implemented");
```
Comment (phildini) [inline batavia/types/Tuple.js]: Would you mind removing this line?

File: batavia/types/Tuple.js (hunk 1 above)
Comment (kennethlove) [inline batavia/types/Tuple.js]: That seems to be a rendering bug in GitHub. I don't have that indentation in my source file.

**Pull request merged**